### PR TITLE
[masterbots.ai] feat: Thread List Display + Components Comments for AI

### PR DESCRIPTION
**Pull request by Bran18**
Components preview: 

### Browse 

https://github.com/user-attachments/assets/bb95b264-8f47-4590-ac1f-b1e11750ef30

### Chat 
https://github.com/user-attachments/assets/7daa2447-01a8-443b-8c77-e0812b8af656

## Summary by Sourcery

Implement a thread list display feature and enhance the user interface by highlighting the active thread in browse and chat sections.

New Features:
- Introduce a thread list display feature to enhance user navigation between different chat threads.

Enhancements:
- Add a new way to highlight the active thread in both browse and chat sections for better user clarity.

## Summary by CodeRabbit

- **New Features**
	- Introduced multiple new components: `BrowseAccordion`, `BrowseCategoryButton`, `BrowseCategoryTabs`, `BrowseChatMessageList`, `BrowseChatMessages`, `BrowseChatbotDetails`, `BrowseSpecificThreadList`, `NewChat`, `PromptForm`, `SuccessContent`, `ShareLink`, `DateRangePicker`, and `ThreadList`.
	- Enhanced chat functionality with a new message list and category buttons for easier navigation.
	- Added date range selection for filtering threads.
	- Implemented sharing functionality for threads.

- **Improvements**
	- Enhanced documentation across components for better clarity.
	- Improved accessibility features and user experience through dynamic styling and feedback mechanisms.

- **Bug Fixes**
	- Streamlined state management and rendering logic for various components to ensure consistent behavior.

- **Refactor**
	- Updated import statements for types to improve clarity and maintainability.
	- Refactored components to enhance readability and structure without changing functionality.

### CODE REVIEW

**Comment by AndlerRL** [comment]
@Bran18 -- First, about the UX :ok_hand: :star_struck: can we do a variant with the accent colours for dark and light themes?

The glow bluish-purple (probably Mirage) can be replaced with the "accent" colours (not defined on tailwind.config due other colours conflict):

- Dark mode: #83E56A (opacity can be low, around 5% to 20%)
- Light mode: #BE17E8 (opacity can be low, around 5% to 20%)

**Comment by Bran18** [comment]
> @Bran18 --

You were 100% right it looks way better with those ones! Done, is ready to go with the new colors

File: apps/masterbots.ai/components/routes/chat/chat-clickable-text.tsx [hunk 1]
```diff
@@ -22,19 +28,23 @@ export function ClickableText({
...
+
+  //* Check if fullText matches unique pattern to set clickableText and restText.
   const uniqueMatch = fullText.match(uniquePattern)
   let clickableText = uniqueMatch ? uniqueMatch[1] : ''
   let restText = uniqueMatch
     ? fullText.slice(fullText.indexOf(clickableText) + clickableText.length)
     : fullText
 
-  // If the UNIQUE pattern isn't found, use the general pattern
+  //* Fallback to general pattern if uniqueMatch is not found.
```
[the remark this answers is not in the document]
Comment (Bran18) [reply apps/masterbots.ai/components/routes/chat/chat-clickable-text.tsx:47]: @coderabbitai im addressing this one as part of the mentioned improvements

**Review by AndlerRL — APPROVED** [review]

**Comment by AndlerRL** [comment]
[Brandon Development Support - November 2024, 1 of 2](https://trello.com/c/uQypMx4x/264-brandon-development-support-november-2024-1-of-2)

**Pull request merged**